### ISSUE TITLE
`goa gen` and `goa example` uses relative paths instead of packages with Go 1.13

### ISSUE THREAD

**Issue by huysamen** [goadesign/goa#2274]
When running `goa gen` or `goa example` on a project while using Go 1.13, the generated files reference servers/clients by relative path, instead of package.

Using the example from the getting started guide, the import of the `calc` service in:
`gen/http/calc/server/server.go` is as follows:

`calc "../../../../path/to/project/gen/calc"`

With no other changes, switching back to Go 1.12 fixes the issue.

**Comment by raphael** [comment]
Thank you for the report! will look into it.

**Comment by xlab** [comment]
Same here, after updating to Go 1.13 and running as always
```
goa gen xxx/api/design -o ..
```

will generate packages with **absolute** paths, like `/Users/xlab/dev/go/...`

**Comment by raphael** [comment]
This is fixed in the head of the v3 branch, will be releasing v3.05 with the fix soon.

**Comment by wayneashleyberry** [comment]
hey @raphael, I just tried out `v3.0.5` against the example project in `README.md` and `goa` is now generating absolute paths.

Here's the output from the example `gen/http/calc/client/cli.go`

```go
// Code generated by goa v3.0.5, DO NOT EDIT.
//
// calc HTTP client CLI support package
//
// Command:
// $ goa gen calcsvc/design

package client

import (
	calc "/Users/wayneberry/Desktop/calcsvc/gen/calc"
	"fmt"
	"strconv"
)
```

**Comment by raphael** [comment]
Thank you, there was an issue with my fix where it did not take into account the case where the top level directory isn't a valid Go package (e.g. no *.go file like when you start a new design from scratch). It needs to be a valid Go package so Goa can leverage the Go parser to compute the package names of the generated code. The code now is more robust as it uses the `gen` folder (guaranteed to be empty) and creates a dummy Go file prior to invoking the Go library to parse it. I've pushed the new fix to `v3` and tagged `v3.0.6`.

**Comment by wayneashleyberry** [comment]
`v3.0.6` did the trick, thanks so much @raphael
